commitserver/commit/hydratorhelper_test.go

```diff
@@ -4,17 +4,38 @@ import (
 	"encoding/json"
 	"os"
 	"path"
+	"path/filepath"
 	"testing"
 
-	securejoin "github.com/cyphar/filepath-securejoin"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/argoproj/argo-cd/v3/commitserver/apiclient"
 )
 
+// tempRoot creates a temporary directory and returns an os.Root object for it.
+// We use this instead of t.TempDir() because OSX does weird things with temp directories, and it triggers
+// the os.Root protections.
```

Comment:
Potentially because on MacOs, `/var/tmp/....` will result in `/private/var/tmp/....` symlink when used as a current working directory